---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ TARGET=httping
 LOCALEDIR=/usr/share/locale
 
 DEBUG=yes
-WFLAGS=-Wall -W -Wextra -pedantic -D_FORTIFY_SOURCE=2
+WFLAGS=-std=gnu11 -Wall -W -Wextra -pedantic -D_FORTIFY_SOURCE=2
 OFLAGS=
 CFLAGS+=$(WFLAGS) $(OFLAGS) -DVERSION=\"$(VERSION)\" -DLOCALEDIR=\"$(LOCALEDIR)\"
 LDFLAGS+=-lm
